src/quo2/screens/token_overview.cljs

```diff
@@ -0,0 +1,55 @@
+(ns quo2.screens.token-overview
+  (:require [quo.react-native :as rn]
+            [quo.previews.preview :as preview]
+            [reagent.core :as reagent]
+            [quo2.components.token-overview :as quo2]
+            [quo.design-system.colors :as colors]))
+
+(def descriptor [{:label   "Token:"
+                  :key     :token
+                  :type    :select
+                  :options [{:key   "SNT"
+                             :value "SNT"}
+                            {:key   "ETH"
+                             :value "ETH"}]}
+
+                 {:label "Account Balance:"
+                  :key   :account-balance
+                  :type  :text}
+                 {:label   "Price:"
+                  :key     :price
+                  :type  :text}
+                 {:label   "Percentage-Increase:"
+                  :key     :percentage-change
+                  :type  :text}
+                 {:label   "Currency:"
+                  :key     :currency
+                  :type    :select
+                  :options [{:key   :usd
+                             :value "$"}
+                            {:key   :eur
+                             :value "€"}]}])
+
+(def eth-token (js/require "../resources/images/tokens/mainnet/ETH.png"))
+(def snt-token (js/require "../resources/images/tokens/mainnet/SNT.png"))
+
+(defn cool-preview []
+  (let [state (reagent/atom {:token  "ETH" :account-balance "3.00" :price "1.00" :percentage-change "-3.0" :currency :usd})]
+    (fn []
+      [rn/view {:margin-bottom 50 :padding 16}
+       [preview/customizer state descriptor]
+       [rn/view {:border :black
+                 :border-width 1
+                 :align-items      :center}
+        [quo2/token-balance (merge @state {:token-img-src (if (= (:token @state) "ETH") eth-token snt-token)})]
```

Comment:
`merge` is used mostly when you have a map as the second parameter, not a finite set of known keys.
In that case you can use `assoc`.
In your case:

`(assoc @state :token-img-src (if ....))`
But it's multivariadic, so you can also use it to `assoc` multiple keys, for example

`(assoc some-map :a 1 :b 2)`, equivalent to `(merge some-map {:a 1 :b 2})`, but clearer and probably a tad faster.

Reply:
thanks for the help :)